pyro/distributions/util.py

```diff
@@ -144,12 +143,12 @@ def sum_leftmost(value, dim):
         return value
     if dim >= value.dim():
         return value.sum()
-    return value.contiguous().view(-1, *value.shape[dim:]).sum(0)
+    return value.reshape(-1, *value.shape[dim:]).sum(0)
```

Comment:
`.reshape` copies the tensor and may be more expensive than `.view`, so it may be advisable to retain the `.view` method.

Reply:
according to http://pytorch.org/docs/master/torch.html#torch.reshape, reshape will not copy when possible
I test it locally and it returns a view for a tensor

Reply:
That's great! Verified that it is fast too. :)